processing/src/main/java/io/druid/segment/FloatWrappingDimensionSelector.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.segment.data.IndexedInts;
+import it.unimi.dsi.fastutil.ints.IntIterator;
+import it.unimi.dsi.fastutil.ints.IntIterators;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class FloatWrappingDimensionSelector implements DimensionSelector
+{
+  private final FloatColumnSelector delegate;
+  private final ExtractionFn exFn;
+  private final Map<String, Integer> valueToId = Maps.newHashMap();
```

Comment:
Agree on having the dictionary management reside in the query engines and other users, I'm in favor of getting #3858 merged first so I'll hold off on changing this part for now